File: Prosolve-6 2018 (National)/Prosolve National - Problem Sets/Test Case(s)/Real/I. Prime Test/solution.py

def prime_test(n):
    if n <= 1: return False
    if n <= 3: return True
    for i in range(2, n-1):
        if n % i == 0:
            return False
    return True

def main():
    primes = []
    for i in range(2, 10001):
        if prime_test(i):
            primes.append(i)

    T = int(input())
    for _ in range(T):
        N = int(input())
        if N in primes:
            print("Prime")
        else:
            for p in primes:
                if N % p == 0:
                    print(p)
                    break

if __name__ == '__main__':
    main()

File: Prosolve-6 2018 (National)/Prosolve National - Problem Sets/Test Case(s)/Real/I. Prime Test/solution.cpp

#include <iostream>
#include <cstdio>
#include <vector>
#include <algorithm>

using namespace std;

bool is_prime(int n)
{
    if (n <= 1) return false;
    if (n <= 3) return true;
    for (int i=2; i < n-1; i++)
        if (n % i == 0)
            return false;
    return true;
}

int main()
{
    vector<int> primes;
    for (int i=2; i <= 10000; i++)
        if (is_prime(i))
            primes.push_back(i);

    int T, N;

    cin >> T;
    for (int i=0; i<T; i++) {
        cin >> N;
        if(std::find(primes.begin(), primes.end(), N) != primes.end()) {
            cout << "Prime" << endl;
        } else {
            for(int j = 0; j != primes.size(); j++) {
                if (N % primes[j] == 0) {
                    cout << primes[j] << endl;
                    break;
                }
            }
        }
    }

    return 0;
}